aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/RetryHandler.kt

```diff
@@ -0,0 +1,51 @@
+package com.amplifyframework.datastore.syncengine
+
+import com.amplifyframework.datastore.utils.ErrorInspector
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.core.SingleEmitter
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+
+class RetryHandler(
+    private val maxExponent: Int = 8,
+    private val jitterFactor: Int = 100,
+    private val maxAttempts: Int = 3
+) {
+
+    private var numberOfAttempts = 0
+
+
+    fun <T> retry(single: Single<T>, skipExceptions: List<Class<out Throwable>?>): Single<T> {
+        return Single.create { emitter -> call(single, emitter, 0, skipExceptions) }
+    }
+
+    private fun <T> call(
+        single: Single<T>,
+        emitter: SingleEmitter<T>,
+        delayInSeconds: Long,
+        skipExceptions: List<Class<out Throwable>?>
+    ) {
+        single.delaySubscription(delayInSeconds, TimeUnit.SECONDS)
+            .subscribe({
+                emitter.onSuccess(it)
+            }) { error ->
+                numberOfAttempts++
+                if (numberOfAttempts > maxAttempts || ErrorInspector.contains(error, skipExceptions)) {
+                    emitter.onError(error)
+                } else {
+                    call(single, emitter, jitteredDelay(), skipExceptions)
+                }
+            }
+    }
+
+
+    fun jitteredDelay(): Long {
+        val waitTimeSeconds: Long =
+            2.0.pow((numberOfAttempts % maxExponent)).toLong()
+        +jitterFactor * Math.random()
+
+        return TimeUnit.MILLISECONDS.toMillis(waitTimeSeconds)
```

Comment:
I think this is converting milliseconds to milliseconds, which is basically not doing anything.  I'd suggest making sure all variable and method names include the units in the name itself, for clarity.  So, change `jitteredDelay()` to `jitteredDelayMillis()` or jitteredDelaySeconds()`, depending on what it is returning.